src/App.tsx

```diff
@@ -1,61 +1,124 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const getUserById = (userId: number) => usersFromServer
+  .find(user => user.id === userId)
+  || null;
+
+const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('');
+  const [hasTitleError, setTitleError] = useState(false);
+  const [hasUserError, setUserError] = useState(false);
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+
+  const createNewTodo = () => {
+    const maxId = Math.max(...visibleTodos.map(todo => todo.id));
```

Comment:
This is relatively heavy. I think you can store the last ID in a variable just to avoid calculating it at each render. Therefore would be calculated only once.

Reply:
Thank you @UmizDemud . I agree with you. I tried to improve it now.